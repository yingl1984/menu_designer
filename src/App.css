html, body {
  margin: 0;
  width: 100%;
  height: 100%;
}

.link {
  color: #264653
}
.link:hover {
  color: #147df5
}

.app {
  display: flex;
  flex-direction: column;
  font-family:-apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen", "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue", sans-serif;
  text-align: left;
  margin:auto;
  width: 50%;
}


.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
