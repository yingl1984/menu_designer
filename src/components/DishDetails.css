/* Flex */
.noteView{
    display: flex;
    flex-direction: column;
    font-family:-apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen", "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue", sans-serif;
    margin:auto;
    width:600px;
}

.group{
    display: flex;
    flex-direction: column;
    font-family:-apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen", "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue", sans-serif;
}

.item{
    flex: 1;
    margin: 20px;
}

/* Form */
input[type=text], input[type=number],select, textarea {
  width: 100%;
  padding: 12px;
  border: 1px solid #ccc;
  border-radius: 4px;
  resize: vertical;
}  
  
label {
  display: inline-block;
}

.container {
  border-radius: 5px;
  background-color: rgba(231, 226, 226, 0.822);
  padding: 20px;
}
  
.col-25 {
  float: left;
  width: 25%;
  margin-top: 6px;
}

.col-75 {
  float: left;
  width: 70%;
  margin-top: 6px;
}

/* Button */
button {
  background-color: #04aa6de8;
  color: white;
  padding: 10px 15px;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  margin-right: 10px;
  margin-top:20px;
}

button:hover {
  background-color: #b8a865;
}