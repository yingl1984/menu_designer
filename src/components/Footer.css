footer{
    display: flex;
    flex-direction: column;
    text-align: right;
    font-size: medium;
    font-family:Cambria, Cochin, Georgia, Times, 'Times New Roman', serif
}