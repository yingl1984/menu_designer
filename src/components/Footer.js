import React, { Component } from 'react';
// import "./Footer.css";
class Footer extends Component {
    render() { 
        return (
            <footer className='footer'>
                <p>&copy; Ying Liu</p>
            </footer>
          );
    }
}
 
export default Footer;