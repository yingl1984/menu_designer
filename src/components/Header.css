header {
    display: flex;
    flex-direction: column;
    text-align: center;
    margin-top:30px;
    margin-bottom: 20px;
    font-family:-apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen", "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue", sans-serif;
}



