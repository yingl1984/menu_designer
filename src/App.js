import React, { Component } from "react";
import Header from "./components/Header";
import Footer from "./components/Footer";
import MenuList from "./pages/MenuList";
import "./App.css";

class App extends Component {
  render(){
    return (
      <main className="app">
        <Header />
          <MenuList />
        <Footer />
      </main>
    );
  }
}

export default App;
