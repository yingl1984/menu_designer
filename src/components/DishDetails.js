import React, { Component } from 'react';
import Header from './Header';
import Footer from './Footer';
import conf from "../config";
import fetchCall from "../helpers/fetchData"
// import './NoteView.css';

class DishDetails extends Component {
  state={
    status:"",
    price:"",
    comments:""
  }
  
  async getData(){
    try{
        const id = this.props.match.params.id;
        let notes = await (await fetchCall(`${conf.noteapi}/${id}`)).json();
        if(!notes)
        {
            throw new Error("Crashed!");
        }
        this.state.comments = ""
        this.setState({
            status:notes.status,
            price:notes.price,
            comments:notes.comments
        });
    }
    catch(err)
        {
         return err.message;
        }
    }

    async componentDidMount() {
      this.getData();
    } 

  updateInput = (e)=>{
    const {name,value} = e.target;
    this.setState({
        [name]: value
    });
  }

  updateInputStatus = async(e)=>{
    e.preventDefault();
    document.getElementById("status").removeAttribute("disabled");
    if(document.getElementById("price")){
      document.getElementById("price").removeAttribute("disabled");
    }
    document.getElementById("comments").removeAttribute("disabled");
  }

  updateNote = async(e) => {
    e.preventDefault();

    const id = this.props.match.params.id;
    const newNote = JSON.stringify({
      "status":this.state.status,
      "price":this.state.price,
      "comments": this.state.comments
    })
    let updatedStatus = await (await fetchCall(`${conf.noteapi}/${id}`, 'PATCH', newNote ));
    if(updatedStatus.ok)
      {
          alert("Note updated!");
      }
    this.getData();
    window.location.replace("/notebook");
  }

  render() {
          
    return (      
      <div className="noteView">
        <Header></Header>
        <div className = "group">
          
          <section className = "item">
            <button onClick={()=>this.props.history.goBack()}>Back</button> 
          </section>
          <section className = "item container">
            
            <form>
            <h2>Dish Details</h2>
              <div className="row">
                <div className="col-25">
                  <label htmlFor="status">Dish Name </label> 
                </div>
                <div className="col-75">
                  <input type="text" id="status" name="status" value={this.state.status} disabled onChange={(e)=>this.updateInput(e)}></input>
                </div>
              </div>

              <div className="row">
                <div className="col-25">
                  <label htmlFor="price">Price </label>
                </div>
                <div className="col-75">
                  <input type="number" id="price" name="price" value={this.state.price} disabled onChange={(e)=>this.updateInput(e)}></input>
                </div>
              </div>

              <div className="row">
                <div className="col-25">
                  <label htmlFor="price">Rate </label>
                </div>
                <div className="col-75">
                  <input type="number" id="price" name="price" value={this.state.price} disabled onChange={(e)=>this.updateInput(e)}></input>
                </div>
              </div>

              <div className="row">
                <div className="col-25">
                  <label htmlFor="comments">Comments </label> 
                </div>
                <div className="col-75">
                  <textarea id="comments" name="comments" rows="5" cols="33" value={this.state.comments} disabled onChange={(e)=>this.updateInput(e)}></textarea>
                </div>
              </div>

              <div className="row">
                <button onClick={e => this.updateInputStatus(e)}>Update</button> 
                <button onClick={e => this.updateNote(e)}>Save</button>
              </div>
            </form>                                
          </section>
        </div>             
          <Footer></Footer>
      </div>
             
          )
        }  
}

export default DishDetails;