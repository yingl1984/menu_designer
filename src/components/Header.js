import React, { Component } from "react";
import {NavLink} from "react-router-dom";
import "./Header.css";

class Header extends Component {
    render() { 
        return (
            <header className="headSection">
                <NavLink to='/' className="link" style={{ textDecoration: "none" }}><h1>Restaurant Menu Designer</h1></NavLink>
            </header>
          );
    }
}
 
export default Header;