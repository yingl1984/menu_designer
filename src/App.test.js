import React from 'react';
import App from './App';
import Footer from './components/Footer'
import Header from './components/Header'
import { shallow } from 'enzyme'
import DishDetails from '../src/components/DishDetails'

it('renders Footer component without crashing', () => {
  const wrapper = shallow(<Footer />)
  expect(wrapper).toMatchSnapshot()
})
it('renders Header component without crashing', () => {
  const wrapper = shallow(<Header />)
  expect(wrapper).toMatchSnapshot()
});
it('renders APP component without crashing', () => {
  const wrapper = shallow(<App />)
  expect(wrapper).toMatchSnapshot()
});
it('renders NoteView component without crashing', () => {
  const wrapper = shallow(<DishDetails />)
  expect(wrapper).toMatchSnapshot()
})