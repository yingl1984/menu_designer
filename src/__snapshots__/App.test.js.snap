// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders APP component without crashing 1`] = `ShallowWrapper {}`;

exports[`renders Footer component without crashing 1`] = `ShallowWrapper {}`;

exports[`renders Header component without crashing 1`] = `ShallowWrapper {}`;

exports[`renders NoteView component without crashing 1`] = `ShallowWrapper {}`;
